trio/_threads.py

```diff
@@ -227,17 +213,6 @@ def _run_fn_as_system_task(cb, fn, *args, trio_token=None):
                 "this thread wasn't created by Trio, pass kwarg trio_token=..."
             )
 
-    # TODO: This is only necessary for compatibility with BlockingTrioPortal.
```

Comment:
I think we might still need this for the case where the user passes the trio_token?

Reply:
I have no idea, I just grepped for BlockingTrioPortal, but I'm not familiar with those APIs. @njsmith Can you please confirm?

Reply:
I think @oremanj is right. The test case would be something like:

```python
async def test():
    trio_token = trio.lowlevel.current_trio_token()
    with pytest.raises(RuntimeError):
        trio.from_thread.run_sync(lambda: None, trio_token=trio_token)
```

If we fail to detect we're calling `from_thread` inside the Trio thread, then this will deadlock.